xsub.go

```diff
@@ -72,7 +88,64 @@ func (xsub *xsubSocket) GetOption(name string) (interface{}, error) {
 
 // SetOption is used to set an option for a socket.
 func (xsub *xsubSocket) SetOption(name string, value interface{}) error {
-	return xsub.sck.SetOption(name, value)
+	err := xsub.sck.SetOption(name, value)
+	if err != nil {
+		return err
+	}
+
+	var (
+		topic []byte
+	)
+
+	switch name {
+	case OptionSubscribe:
+		k := value.(string)
+		xsub.subscribe(k, 1)
+		topic = append([]byte{1}, k...)
+
+	case OptionUnsubscribe:
+		k := value.(string)
+		topic = append([]byte{0}, k...)
+		xsub.subscribe(k, 0)
```

Comment:
any reason why in one case it's `subscribe` then `append` while in the other case it's the reverse?

Reply:
Just an oversight. Committed change in issue-108.